docs/contributing.md

```diff
@@ -42,3 +42,93 @@ To build the documentation locally, run the following:
     make livehtml
 
 This will start a live preview server, using [sphinx-autobuild](https://pypi.org/project/sphinx-autobuild/).
+
+## Using Docker Compose
+
+If you're familiar with Docker--or even if you're not--you may awnt to consider using our optional Docker Compose setup.
+
+An advantage of this approach is that it relieves you of setting up any dependencies, such as ensuring that you have the proper version of Python and Postgres and so forth.  On the downside, however, it does require you to familiarize yourself with Docker, which, while relatively easy to use, still has its own learning curve.
+
+To try out the Docker Compose setup, you will first want to [get Docker][] and [install Docker Compose][].
+
+Then, after checking out the code, run the following:
+
+```
+cd django-sql-dashboard
+docker-compose build
+```
+
+At this point, you can start editing code.  To run any development tools such as `pytest` or `black`, just prefix everything with `docker-compose run app`.  For instance, to run the test suite, run:
+
+```
+docker-compose run app python pytest
+```
+
+If this is a hassle, you can instead run a bash shell inside your container:
+
+```
+docker-compose run app bash
+```
+
+At this point, you'll be in a bash shell inside your container, and can run development tools directly.
+
+[get Docker]: https://docs.docker.com/get-docker/
+[install Docker Compose]: https://docs.docker.com/compose/install/
+
+### Using the dashboard interactively
+
+The Docker Compose setup is configured to run a simple test project that you can use to tinker with the dashboard interactively.
+
+To set this up, first run:
+
+```
+docker-compose run app python test_project/manage.py migrate
+docker-compose run app python test_project/manage.py createsuperuser
+```
+
+You will now be prompted to enter details about a new superuser. Once you've done that, run:
+
+```
+docker-compose up
+```
+
+You can now visit the example app's dashboard at http://localhost:8000/dashboard/.
```

Comment:
@simonw I would really like to tell the user to visit http://localhost:8000/ here, but the root of the test project is currently a 404.  Do you want me to add a redirect from `/` to `/dashboard/` or should I keep the URL above?

Reply:
Yes adding that redirect is a good idea